metricflow/specs/query_interface.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+from typing import Sequence
+from abc import ABC
+
+
+class QueryInterfaceMetric(ABC):
```

Comment:
Can we make these Protocol types instead of ABC?

Generally speaking, the Protocol is what we'd use for interfaces - especially cross-codebase interfaces, since inheritance is not required.

Reply:
Can you go into more detail about what you mean by

> inheritance is not required

I am thinking that different parameter types (i.e. group_by, where, order) would inherit from these dimensions, time dimension, and entity interfaces.